Speed_Sensor.ino

```diff
@@ -0,0 +1,46 @@
+// digital pin 2 is the hall pin
+int hall_pin = 2;
+// set number of hall trips for RPM reading (higher improves accuracy)
+float hall_thresh = 100.0;
+
+void setup() {
+  // initialize serial communication at 9600 bits per second:
+  Serial.begin(115200);
+  // make the hall pin an input:
+  pinMode(hall_pin, INPUT);
+}
+
+
+void loop() {
+  // preallocate values for tach
+  float hall_count = 1.0;
+  float start = micros();
+  bool on_state = false;
+  // counting number of times the hall sensor is tripped
+  // but without double counting during the same trip
+  while(true){
+    if (digitalRead(hall_pin)==0){
+      if (on_state==false){
+        on_state = true;
+        hall_count+=1.0;
+      }
+    } else{
+      on_state = false;
+    }
+    
+    if (hall_count>=hall_thresh){

```

Comment:
The `hall_count>=hall_thresh` logic could be put in the while loop condition for better readability.